components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/model/Response.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.registration.engine.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Model class to represent the response of a node in the registration sequence.
+ */
+public class Response {
+
+    private String status;
+    private String type;
+    private final List<String> requiredData =  new ArrayList<>();
+    private Map<String, String> additionalInfo;
+
+    private Response(Builder builder) {
+
+        this.status = builder.status;
+        this.type = builder.type;
+        if (builder.requiredData != null) {
```

Comment:
do we need this check if the requiredData is not initialized in L32. Instead we can assign the value coming from the builder. Does having a initialized list for requiredData serve any purpose?